tests/e2e/conftest.py

```diff
@@ -37,10 +37,9 @@ def pytest_sessionstart():
     # OLS_URL env only needs to be set when running against a local ols instance,
     # when ols is run against a cluster the url is retrieved from the cluster.
     ols_url = os.getenv("OLS_URL", "")
+    pytest.on_cluster = False
     if "localhost" not in ols_url:
         pytest.on_cluster = True
-
-    if pytest.on_cluster:
```

Comment:
Why do we remove this check? If we remove it, we should then not raise an error if the installation fails no?

Reply:
there's a few extra stuff necessary, like declaring the on_cluster variable as global which the pytest.on_cluster cannot do.
This was what was causing the sessionfinish to not run the must_gather